src/ElasticaService.php

```diff
@@ -471,4 +464,52 @@ protected function exception($exception)
         );
         $this->logger->error($message);
     }
+
+    /**
+     * Check if response has any errors
+     *
+     * @param Response|null $response
+     * @throws Exception
+     */
+    protected function logResponse(Response $response = null)
+    {
+        // Ignore empty or non-error responses
+        if (!$response || $response->isOk()) {
+            return;
+        }
+
+        // Get error message
+        $data = $response->getData();
+        $errorMessage = $data['message']
+            ?? $response->getErrorMessage()
+                ?: sprintf("HTTP %d error", $response->getStatus());
+        $message = "Elastica server error: $errorMessage";
+
+        // If no logger specified expose error normally
+        if (!$this->logger) {
+            throw new Exception($message, $response->getStatus());
```

Comment:
Would be good to use module-specific exceptions to better be able to filter in error-aggregation systems like Sentry. e.g. `ElasticSearchException`.

Reply:
Might leave this for later, seperate from the upgrade work.